staging/src/k8s.io/client-go/discovery/cached/memcache.go

```diff
@@ -39,31 +45,46 @@ type memCacheClient struct {
 	delegate discovery.DiscoveryInterface
 
 	lock                   sync.RWMutex
-	groupToServerResources map[string]*metav1.APIResourceList
+	groupToServerResources map[string]*cacheEntry
 	groupList              *metav1.APIGroupList
 	cacheValid             bool
 }
 
 // Error Constants
 var (
-	ErrCacheEmpty    = errors.New("the cache has not been filled yet")
 	ErrCacheNotFound = errors.New("not found")
 )
 
 var _ discovery.CachedDiscoveryInterface = &memCacheClient{}
 
+func isRetryableError(err error) bool {
+	return errorsutil.IsServiceUnavailable(err) || errorsutil.IsInternalError(err) || errorsutil.IsServerTimeout(err) || errorsutil.IsTimeout(err)
```

Comment:
I'm having trouble deciding what types of error are retryable and if the list here is complete. Could you share your thought process?

Reply:
I was trying to catch some server-side errors. My intent was to create potentially incomplete list rather than list with too many errors causing too many retries.

I changed the logic a little bit there to make complete and easier to understand. Right now I retry one of three examples:
* Connection errors (like "Connection refused" "Connection reset" errors)
* http 500+ errors
* 429 Too many requests